duktape/src/main/jni/java/JavaType.cpp

```diff
@@ -16,7 +16,29 @@
 #include "JavaType.h"
 #include "JString.h"
 #include "JavaExceptions.h"
-#include "GlobalRef.h"
+
+jarray JavaType::popArray(duk_context* ctx, JNIEnv* env, uint32_t count, bool inScript) const {
+  jobjectArray array = env->NewObjectArray(count, static_cast<jclass>(m_classRef.get()), nullptr);
+  for (auto i = count; i > 0u; --i) {
+    env->SetObjectArrayElement(array, i - 1, pop(ctx, env, inScript).l);
+  }
+  return array;
+}
+
+duk_ret_t JavaType::pushArray(duk_context* ctx, JNIEnv* env, const jarray& values) const {
+  const auto size = env->GetArrayLength(values);
+  jvalue element;
```

Comment:
Habit/premature optimization. Some mystics believe that if you declare it inside the loop, the compiler will keep allocating and deallocating it on each iteration, when we really just want a place to update our `.l` each time.  It's not a big struct, though, so it's probably not worth the hassle.